utilities/checkpoint/checkpoint_impl.cc

```diff
@@ -229,13 +229,13 @@ Status CheckpointImpl::CreateCustomCheckpoint(
     }
   }
 
-  // this will return live_files prefixed with "/"
-  s = db_->GetLiveFiles(live_files, &manifest_file_size, flush_memtable);
-
   if (!db_->GetIntProperty(DB::Properties::kMinLogNumberToKeep, &min_log_num)) {
     return Status::InvalidArgument("cannot get the min log number to keep.");
   }
 
+  // this will return live_files prefixed with "/"
+  s = db_->GetLiveFiles(live_files, &manifest_file_size, flush_memtable);
```

Comment:
If GetLiveFiles() is called after getting kMinLogNumberToKeep, how do we guarantee all files after min_log_num is preserved and returned by GetLiveFiles()?

Reply:
we do not need to make this guarantee. If some WALs after min_log_num  do not show up in GetLiveFiles(), then they won't show up in checkpoint anyway.

Reply:
we only make sure that the WALs that are "alive" according to MANIFEST really show up in the checkpoint.

Reply:
Oh GetLiveFiles() doesn't fetch log files. Ignore te question then.